lib/runtime2/threepio/utils/llmClient.ts

```diff
@@ -1,107 +1,65 @@
+  input: string,
+  systemPrompt: string,
 }
 
-interface MonitoredLLMCallData {
-  startTime: number;
-  endTime: number;
-  durationMs: number;
-  input: string;
-  systemPrompt: string;
-  jsonSegments: JsonSegmentTiming[];
-  errors?: string[]; // Added errors array
-}
-
-export async function* callLLM(input: string, systemPrompt: string): ApiStream {
+export function callLLM(options: CallLLMOptions): { stream: ApiStream, requestId: string } {
+  const { input, systemPrompt } = options;
+  const requestId = input.substring(0, 10) + Date.now();
   const config: ApiConfiguration = {
     apiProvider: getThreepioApiProvider() as ApiProvider,
-    apiModelId: getThreepioApiModelId(), // Corrected typo: removed extra 'n'
+    apiModelId: getThreepioApiModelId(),
     apiKey: getThreepioApiKey(),
     endpoint: getThreepioApiEndpoint(),
   };
   const handler = buildApiHandler(config);
-  const callStartTime = Date.now();
-  const monitoringData: MonitoredLLMCallData = {
-    startTime: callStartTime,
-    systemPrompt: "systemPrompt",
-    input: input,
-    jsonSegments: [],
-    errors: [],
-    endTime: 0,
-    durationMs: 0
-  };
-
-  const originalStream = handler.createMessage(systemPrompt, [{ role: 'user', content: input }]);
-  const processedStream = processJsonlStream(originalStream, monitoringData, '');
-  for await (const chunk of processedStream) {
-    yield chunk;
+  try {
+    const originalStream = handler.createMessage(systemPrompt, [{ role: 'user', content: input }]);
+    const processedStream = processJsonlStream(originalStream, requestId);
+    return { stream: processedStream, requestId }; // Return the callId for downstream use
+  } catch (error: any) {
+    reportThreepioError(error.message, requestId);
   }
 }
 
-function updateMonitoringForJsonSegment(
-  processedLine: ProcessedJsonLine,
-  monitoringData: MonitoredLLMCallData,
-  lastSegmentTimeRef: { time: number }
-): void {
-  const currentTime = Date.now();
-  const duration = currentTime - lastSegmentTimeRef.time;
-  lastSegmentTimeRef.time = currentTime;
-
-  monitoringData.jsonSegments.push({
-    jsonContent: processedLine.jsonContent || processedLine.error || {},
-    rawLine: processedLine.rawLine,
-    startTime: currentTime - duration,
-    endTime: currentTime,
-    durationMs: duration,
-  });
-}
-
 async function* processJsonlStream(
   sourceStream: ApiStream,
-  monitoringData: MonitoredLLMCallData,
-  spanIdForLogging: string
+  requestId: string
 ): ApiStream {
-  const jsonlProcessor = new JsonlStreamProcessor();
-  const lastSegmentTimeRef = { time: monitoringData.startTime };
-
+  const jsonlProcessor = new JSONLProcessor();
   try {
     for await (const sourceChunk of sourceStream) {
       if (sourceChunk.type === 'text') {
         for (const processedLine of jsonlProcessor.processChunk(sourceChunk.text)) {
+          if (isDebugging) {
+            reportThreepioInfo('Debug:Processing text chunk', sourceChunk.text, 'requestId', requestId);
```

Comment:
为什么不直接在 `if (sourceChunk.type == 'text')` 打印，这样会丢消息（如果有其他 type）

另外，打印信息应该改成：

```suggestion
            reportThreepioInfo('Source chunk:', { chunk: sourceChunk, requestId });
```

1. 这里的目的不是为了 Debug 处理而是获取上游返回数据，即 sourceChunk
2. 使用对象更易读